lib/tasks/bundle_eval.rake

```diff
@@ -0,0 +1,194 @@
+
+              calc_diffs = false
+              effective_date_end = Time.at(bundle.effective_date).in_time_zone.to_formatted_s(:number)
+              effective_date = Time.at(bundle.measure_period_start).in_time_zone.to_formatted_s(:number)
+              options = { 'effectiveDateEnd': effective_date_end, 'effectiveDate': effective_date }
+              cw_patient = patient.clone
+              cw_patient.save
+              original_results = SingleMeasureCalculationJob.perform_now([cw_patient.id.to_s], measure.id.to_s, bundle.id.to_s, options)
+              codes.each_value do |cw|
+                next if calc_diffs
+
+                cw_patient.qdmPatient.dataElements[de_index].dataElementCodes = [{ 'code' => cw[:code], 'codeSystemOid' => cw[:code_system] }]
+                cw_patient.save
+                new_results = SingleMeasureCalculationJob.perform_now([cw_patient.id.to_s], measure.id.to_s, bundle.id.to_s, options)
+                calc_diffs = are_results_different?(original_results, new_results)
+              end
+              cw_patient.delete
+              csv << [measure.cms_id, "#{patient.givenNames[0]} #{patient.familyName}", de_index, data_element._type, codes] if calc_diffs
+            end
+          end
+        end
+      end
+    end
+
+    desc %(Compares the calculation results across 2 bundles.  Presumably, both bundles share the same test patients.
+    When a difference is found, the compare_calculations.csv will include the patient name and cms_id with differences.
+
+    You must identify two bundle versions:
+
+    $ rake bundle:eval:compare_calculations[xxxx.x.x,xxxx.x.x])
+    task :compare_calculations, %i[bundle_version_1 bundle_version_2] => :setup do |_, args|
+      first_bundle = Bundle.where(version: args.bundle_version_1).first
+      second_bundle = Bundle.where(version: args.bundle_version_2).first
+      CSV.open('tmp/compare_calculations.csv', 'w', col_sep: '|') do |csv|
+        # Iterate through all patients in both bundles to catch instances where calculations may be nonexistent on one of the bundles
+        first_bundle.patients.each do |first_patient|
+          second_patient = second_bundle.patients.where(givenNames: first_patient.givenNames, familyName: first_patient.familyName).first
+          # Skip if the same patient can't be found in both bundles
+          next unless second_patient
+
+          first_patient.calculation_results.each do |fp_cr|
+            measure_id = second_bundle.measures.where(cms_id: fp_cr.measure.cms_id).first.id
+            sp_cr = second_patient.calculation_results.where(measure_id: measure_id, population_set_key: fp_cr.population_set_key).first
+            calc_diffs = sp_cr.nil? ? true : false
+            calc_diffs ||= result_different?(fp_cr, sp_cr)
+            csv << [fp_cr.measure.cms_id, first_patient.givenNames[0], first_patient.familyName] if calc_diffs
+          end
+        end
+        second_bundle.patients.each do |second_patient|
```

Comment:
I'll put a comment here.  
Since we're looping through Individual Results (which are only stored when the IPP > 0) if we only look at the first bundle, we could miss a case where that same patient is not in the IPP in bundle 1, but is in the IPP in bundle 2.